src/ProjectTemplates/BlazorWasm.ProjectTemplates/BlazorWasm-CSharp.Client.csproj.in

```diff
@@ -11,10 +11,25 @@
     <PackageReference Include="Microsoft.AspNetCore.Blazor.DevServer" Version="${MicrosoftAspNetCoreBlazorDevServerPackageVersion}" PrivateAssets="all" />
     <PackageReference Include="Microsoft.AspNetCore.Blazor.HttpClient" Version="${MicrosoftAspNetCoreBlazorHttpClientPackageVersion}" />
   </ItemGroup>
+
   <!--#if Hosted -->
   <ItemGroup>
     <ProjectReference Include="..\Shared\BlazorWasm-CSharp.Shared.csproj" />
   </ItemGroup>
+
   <!--#endif -->
+  <!--#if PWA -->
+  <Target Name="PublishServiceWorker" AfterTargets="CopyFilesToPublishDirectory">
+    <!-- Use the "published" version of the service worker in the output -->
+    <PropertyGroup>
+      <ServiceWorkerMoveFrom>$(PublishDir)$(BlazorPublishDistDir)service-worker.published.js</ServiceWorkerMoveFrom>
+      <ServiceWorkerMoveTo>$(PublishDir)$(BlazorPublishDistDir)service-worker.js</ServiceWorkerMoveTo>
+    </PropertyGroup>
+    <Move SourceFiles="$(ServiceWorkerMoveFrom)" DestinationFiles="$(ServiceWorkerMoveTo)" />
 
+    <!-- Append a unique version number that will trigger installation of updates -->
+    <WriteLinesToFile File="$(ServiceWorkerMoveTo)" Lines="const serviceWorkerVersion = '$([System.Guid]::NewGuid())'" />
```

Comment:
I don't have any reason to think a timestamp would be preferable, but one of the goals here is that people are free to change this in any way they want. The arrangement provided in the template is just one possible example of how it can be achieved already, independently of any features we ship.

Reply:
Appending a distinct value is what causes the byte-for-byte comparison to recognize this is a new deployment.

Reply:
I'm not sure what I'm missing here. If the bytes are the same, why do we need an update? Doesn't that mean nothing changed in your SW/PWA?

Reply:
Because the thing that changed may be in a file *other* than `service-worker.js`. For example, the assets list that we download from `service-worker.js`.

Reply:
Actually if we import the assets list via `self.importScripts` then it will be sufficient to have a change in the assets list. This fractionally complicates the format of the assets list, but it's probably worth it for the avoidance of having to modify the service worker file on publish.

Reply:
It's now changed to use `self.importScripts`. We still do need a version string in order to produce independent caches per installation. The reasons why that's imported are noted in a separate comment.

I didn't know before that using `self.importScripts` was enough to make the browser recognize updates when you've modified one of those imported scripts. Apparently that's a relatively new feature [since Chrome 78 in September last year](https://developers.google.com/web/updates/2019/09/fresher-sw), though other browsers have supported it for longer.